### PR TITLE
Add Watcher3 module

### PR DESCRIPTION
**Pull request by scambra**

### CODE REVIEW

**Comment by scambra** [comment]
@MichaIng you should be able to edit files

**Comment by MichaIng** [comment]
Many thanks, yes that worked 👍.

**Comment by gmiranda** [comment]
There's an error on my test machine:
-            importlib.reload(sys)
+            reload(sys)

@MichaIng this PR seems to be based on old code, isn't it?

**Comment by MichaIng** [comment]
Yes the original base branch was Hellowlol's Python 2 one, so this needs to be updated for Python 3.

File: htpc/helpers.py [hunk 1]
```diff
@@ -235,11 +235,14 @@ def download_image(url, dest, auth=None, headers=None):
...
-
+        resp = urlopen(request)
+        body = resp.read()
+        if resp.info().type == 'application/json':
+            logger.error('Failed to download %s to %s: %s' % (url, dest, body))
+            return
+        elif body:
             with open(dest, 'wb') as local_file:
-                local_file.write(urlopen(request).read())
+                local_file.write(body)
```
Comment (MichaIng) [inline htpc/helpers.py:245]: This is not very generic and probably there is a better way to check for soft 404 returns. But a JSON return is definitely not an image, so this does not cause any issues.

A bonus is the removed doubled `urlopen`.

**Comment by MichaIng** [comment]
@scambra 
If you are familiar with Git, could you rebase the branch onto our current `dev` branch? That would make testing easier. I tried it, but while I have push permissions via GitHub, the Git command line utility returns a 403 when trying to push to your repository.

**Comment by scambra** [comment]
I have rebased and added new PR #47

**Pull request closed without merging**